spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/BlasWrapper.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn.mkldnn
+
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.mkl.Memory
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity, TensorModule}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Shape
+import spire.syntax.module
+
+// only support wrap bigdl to support nchw or nc
+private[bigdl] class BlasWrapper(val module: TensorModule[Float]) extends MklDnnLayer {
```

Comment:
private val module ?

Reply:
and why it's hard-coded to float ? what if int or fp16?

Reply:
Now BlasWrapper is only used when converting to dnn graph. As tests are going on, I think not expose BlasWrapper  to users is better.

Reply:
As for  hard-coded to float, dnn layer now only supports float tensor, so here tensor data type should be float.
For int or fp16 type, I am not sure how to define the model.

Reply:
I mean the parameter `module`  should be private, right ?

Reply:
In future, we may have in8, fp16 training/inference, so probably we need to support them, just make it generic ?